shell/platform/fuchsia/flutter/tests/flatland_platform_view_unittest.cc

```diff
@@ -0,0 +1,1358 @@
+  response4->ExpectCompleted("");
+  response5->ExpectCompleted("");
+  response6->ExpectCompleted("");
+}
+
+// This test makes sure that the PlatformView correctly returns a Surface
+// instance that can surface the provided gr_context and external_view_embedder.
+TEST_F(FlatlandPlatformViewTests, CreateSurfaceTest) {
+  MockPlatformViewDelegate delegate;
+
+  flutter::TaskRunners task_runners =
+      flutter::TaskRunners("test_runners",  // label
+                           nullptr,         // platform
+                           flutter_runner::CreateFMLTaskRunner(
+                               async_get_default_dispatcher()),  // raster
+                           nullptr,                              // ui
+                           nullptr                               // io
+      );
+
+  // Test create surface callback function.
+  sk_sp<GrDirectContext> gr_context = GrDirectContext::MakeMock(
+      nullptr,
+      flutter::MakeDefaultContextOptions(flutter::ContextType::kRender));
+  std::shared_ptr<MockExternalViewEmbedder> external_view_embedder =
+      std::make_shared<MockExternalViewEmbedder>();
+  auto CreateSurfaceCallback = [&external_view_embedder, gr_context]() {
+    return std::make_unique<flutter_runner::Surface>(
+        "PlatformViewTest", external_view_embedder, gr_context.get());
+  };
+
+  auto platform_view = PlatformViewBuilder(delegate, std::move(task_runners))
+                           .SetCreateSurfaceCallback(CreateSurfaceCallback)
+                           .SetExternalViewEmbedder(external_view_embedder)
+                           .Build();
+  platform_view.NotifyCreated();
+
+  RunLoopUntilIdle();
+
+  EXPECT_EQ(gr_context.get(), delegate.surface()->GetContext());
+  EXPECT_EQ(external_view_embedder.get(),
+            platform_view.CreateExternalViewEmbedder().get());
+}
+
+// This test makes sure that the PlatformView correctly registers Scenic
+// MetricsEvents sent to it via FIDL, correctly parses the metrics it receives,
+// and calls the SetViewportMetrics callback with the appropriate parameters.
+TEST_F(FlatlandPlatformViewTests, SetViewportMetrics) {
+  constexpr uint32_t width = 640;
+  constexpr uint32_t height = 480;
+
+  MockPlatformViewDelegate delegate;
+  EXPECT_EQ(delegate.metrics(), flutter::ViewportMetrics());
+
+  MockParentViewportWatcher watcher;
+  std::vector<fuchsia::ui::scenic::Event> events;
+  flutter::TaskRunners task_runners("test_runners", nullptr, nullptr, nullptr,
+                                    nullptr);
+  auto platform_view = PlatformViewBuilder(delegate, std::move(task_runners))
+                           .SetParentViewportWatcher(watcher.GetHandle())
+                           .Build();
+  RunLoopUntilIdle();
+  EXPECT_EQ(delegate.metrics(), flutter::ViewportMetrics());
+
+  watcher.SetLayout(width, height);
+  RunLoopUntilIdle();
+  EXPECT_EQ(delegate.metrics(),
+            flutter::ViewportMetrics(1.0, width, height, -1.0));
```

Comment:
Naming the fields can help those unfamiliar. 
/*DPR*/ 1.0, width, height, /*slop*/ -1.0)